js/map.js

```diff
@@ -44,10 +37,51 @@
     }
   };
 
+  var openPopup = function () {
+    mapBlock.addEventListener('click', function (evt) {
+      var mapBlockChildren = mapBlock.children;
+      var target = evt.target;
+      while (!target.classList.contains('map')) {
+        if (target.classList.contains('popup__close')) {
+          closePopup();
+          break;
+        }
+        if (target.classList.contains('map__pin') && !target.classList.contains('map__pin--main')) {
+          for (var i = 0; i < mapBlockChildren.length; i++) {
+            if (mapBlockChildren[i].classList.contains('map__card')) {
+              closePopup();
+            }
+          }
+          renderCard(dataList, target.name);
+          document.addEventListener('keydown', onEscClosePopup);
+          break;
+        }
+        target = target.parentNode;
+      }
+    });
+  };
+
+  var closePopup = function () {
+    mapBlock.removeChild(document.querySelector('.map__card'));
+  };
+
+  var onEscClosePopup = function (evt) {
+    evt.preventDefault();
+    window.util.isEscEvent(evt, closePopup);
+    document.removeEventListener('keydown', onEscClosePopup);
+  };
+
+  var onSuccess = function (data) {
+    renderPins(data);
+    dataList = data;
+    openPopup();
+  };
+
   housingTypeFilter.addEventListener('change', function () {
-    for (var i = 0; i < mapPins.length; i++) {
-      if (!mapPins[i].classList.contains('map__pin--main')) {
-        pinList.removeChild(mapPins[i]);
+    var delPins = document.querySelectorAll('.map__pin');
```

Comment:
```suggestion
    var delPins = document.querySelectorAll('.map__pin:not(.map__pin--main)');
```
тогда условие в цикле не нужно